WalletWasabi.Fluent/Controls/PrivacyBar.axaml.cs

```diff
@@ -0,0 +1,91 @@
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Generators;
+using Avalonia.Controls.Presenters;
+using Avalonia.VisualTree;
+using System.Linq;
+
+namespace WalletWasabi.Fluent.Controls;
+
+public class PrivacyBar : ItemsPresenter
+{
+	private const double GapBetweenSegments = 1.5;
+	private const double EnlargeThreshold = 2;
+	private const double EnlargeBy = 1;
+
+	public static readonly StyledProperty<decimal> TotalAmountProperty =
+		AvaloniaProperty.Register<PrivacyBarSegment, decimal>(nameof(TotalAmount));
+
+	public decimal TotalAmount
+	{
+		get => GetValue(TotalAmountProperty);
+		set => SetValue(TotalAmountProperty, value);
+	}
+
+	protected override IItemContainerGenerator CreateItemContainerGenerator()
+	{
+		return new PrivacyBarItemContainerGenerator(this);
+	}
+
+	protected override Size ArrangeOverride(Size finalSize)
+	{
+		var children =
```

Comment:
There is a lot of linq, similar to my comment on PrivacyBarSegment, maybe we can later refactor it to simple drawing using Render method, would save a lot memory and perf I think. We can try doing that later.